pkgs/development/libraries/vte/default.nix

```
@@ -73,9 +74,10 @@ stdenv.mkDerivation rec {
     systemd
   ];
 
-  propagatedBuildInputs = assert (gtkVersion == "3" || gtkVersion == "4"); [
+  propagatedBuildInputs = assert (gtkVersion == "3" || gtkVersion == "4" || gtkVersion == null);
     # Required by vte-2.91.pc.
-    (if gtkVersion == "3" then gtk3 else gtk4)
+    lib.optional (gtkVersion != null) (if gtkVersion == "3" then gtk3 else gtk4)
+  ++ [
```

Maybe something like this would be cleaner:

```suggestion
    # Required by vte-2.91.pc.
    propagatedBuildInputs = lib.optionals (gtkVersion != null) [
      (assert (gtkVersion == "3" || gtkVersion == "4";
      if gtkVersion == "3" then gtk3 else gtk4)
    ] ++ [
```